libraries/aws/provisioning/src/aws_iot_provisioning_parser.c

```diff
@@ -629,17 +629,20 @@ AwsIotProvisioningError_t _AwsIotProvisioning_ParseRegisterThingResponse( AwsIot
                     IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_PROVISIONING_INTERNAL_FAILURE );
                 }
 
-                pDeviceConfigurationList = AwsIotProvisioning_MallocDeviceConfigurationList( numOfDeviceConfigurationEntries *
+                if ( numOfDeviceConfigurationEntries > 0) {
+                     pDeviceConfigurationList = AwsIotProvisioning_MallocDeviceConfigurationList( numOfDeviceConfigurationEntries *
                                                                                              sizeof( AwsIotProvisioningResponseDeviceConfigurationEntry_t ) );
 
-                if( pDeviceConfigurationList == NULL )
-                {
-                    IotLogError( "Failure in allocating memory for device configuration data in response payload of %s operation",
+                    if( pDeviceConfigurationList == NULL )
+                    {
+                        IotLogError( "Failure in allocating memory for device configuration data in response payload of %s operation",
                                  REGISTER_THING_OPERATION_LOG );
```

Comment:
```suggestion
                                     REGISTER_THING_OPERATION_LOG );
```

Reply:
@archigup I am not sure what to do here! Do I just select `commit suggestion` to make the changes?

Reply:
@deepsworld Yes, you can use `commit suggestion` to add the changes, and that will apply the formatting changes. Thanks!